lib/api/src/externals/function.rs

```diff
@@ -126,7 +129,7 @@ impl Function {
     pub fn new_with_env<F, Env>(store: &Store, ty: &FunctionType, env: Env, func: F) -> Self
     where
         F: Fn(&mut Env, &[Val]) -> Result<Vec<Val>, RuntimeError> + 'static,
-        Env: Sized + 'static,
+        Env: Sized + WasmerEnv + 'static,
```

Comment:
Can we have here Env as `dyn Any`, and then a check on WasmerEnv at runtime?

Reply:
I believe it's not possible to downcast from one trait object into another: it requires downcasting into a concrete type.  Also I think it's just a limitation of the trait system, but there aren't any workarounds for this either.  What I mean is that you can'1t implement a trait on all types but have specializations on certain types (this would let us implement `WasmerEnv` on everything, or another trait which would let us know if `WasmerEnv` was implemented).